File: frontend/lib/graphql/queries/partner.dart
const String getPartners = r'''
query getPartners($partnerId: String){
    getPartners(partnerId: $partnerId){
        id,
        code,
        type,
        active,
        name,
        vat,
        vat_number,
        registration_number,
        address{
            address,
            locality,
            county_code
        },
        bank_account{
            bank,
            iban
        }
    }
}
''';

const String getPartnerByTaxId = r'''
query  getPartnerByTaxId($taxId: String!) {
   getPartnerByTaxId(taxId: $taxId){
        id,
        code,
        name,
        type,
        tax_id,
        company_number,
        address,
        personal_id,
        is_active
    }
}
''';

File: backend/models/partner.go
package models

type Partner struct {
	ID                 string       `json:"id"`
	Code               *string      `json:"code"`
	Type               string       `json:"type"`
	Active             bool         `json:"active"`
	Name               string       `json:"name"`
	VatNumber          *string      `json:"vat_number"`
	Vat                bool         `json:"vat"`
	RegistrationNumber *string      `json:"registration_number"`
	Address            *Address     `json:"company_address"`
	BankAccount        *BankAccount `json:"bank_account"`
}

const (
	CompanyType    = "Persoana Juridica"
	IndividualType = "Persoana Fizica"
)

type BankAccount struct {
	Bank *string `json:"bank"`
	Iban *string `json:"iban"`
}
